test/test.c

```diff
@@ -0,0 +1,128 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include <sys/wait.h>
+
+#define COMMAND_LEN  512
+#define TESTCASE_LEN 32
+#define OUTPUT_LEN   1024
+
+typedef struct testcase_ {
+   char *name;
+   char *argv;
+   char *input;
+   char *output;
+} testcase;
+
+void raise_err(char *err_msg);
+
+int main(int argc, char **argv) {
+   if (argc != 2)
+      raise_err(
+         "test: usage:\n"
+         "\t./test\n"
+         "\t\tPATH_PROGRAM_SOURCE_CODE");
+   if (!system(NULL))
+      raise_err("test: no command processor available.");
+
+   /* Build a program to be tested. */
+   int statcode, retval;
+   char *command;
+   
+   command = malloc(COMMAND_LEN);
+   if (command == NULL)
+      raise_err("test: failed to malloc.");
+
+   /* Note: compile differently as per implementation? */
+   retval = snprintf(command, COMMAND_LEN, "gcc -o object_program %s", argv[1]);
```

Comment:
This can be problem if we have to compile a complicated program.
How about `make all`? #15